velox/core/PlanFragment.h

```diff
@@ -42,10 +43,22 @@ struct PlanFragment {
   ExecutionStrategy executionStrategy{ExecutionStrategy::kUngrouped};
   int numSplitGroups{0};
 
+  /// Contains plan nodes that need to be executed in the grouped mode.
+  std::unordered_set<PlanNodeId> groupedExecutionScanNodeIds;
```

Comment:
Should we replace "scan" with "leaf" or "source" since this can possibly apply to query plans that use custom source node which is not TableScan?